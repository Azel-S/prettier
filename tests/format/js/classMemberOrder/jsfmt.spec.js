run_spec(__dirname, ["babel", "flow", "typescript"], {classMemberOrder:"none"});
run_spec(__dirname, ["babel", "flow", "typescript"], {classMemberOrder:"public first"});
run_spec(__dirname, ["babel", "flow", "typescript"], {classMemberOrder:"private first"});