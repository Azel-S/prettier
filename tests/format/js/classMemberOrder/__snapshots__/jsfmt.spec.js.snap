// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classMemberOrder.js - {"classMemberOrder":"none"} format 1`] = `
====================================options=====================================
classMemberOrder: "none"
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class TestClass {
    publicFieldFirst = 'Public field first'; // Public field
    #privateField = 'Private Field'; // Simulated private field using naming convention
    publicFieldLast = 'Public field last'; // Public field
    #privateFieldLast = 'Private Field Last'; // Simulated private field

    constructor() {
        this.publicFieldConstructor = 'Public field set in constructor';
        this._privateFieldConstructor = 'Private field set in constructor';
        console.log('Constructor');
    }
}
=====================================output=====================================
class TestClass {
  publicFieldFirst = "Public field first"; // Public field
  #privateField = "Private Field"; // Simulated private field using naming convention
  publicFieldLast = "Public field last"; // Public field
  #privateFieldLast = "Private Field Last"; // Simulated private field

  constructor() {
    this.publicFieldConstructor = "Public field set in constructor";
    this._privateFieldConstructor = "Private field set in constructor";
    console.log("Constructor");
  }
}

================================================================================
`;

exports[`classMemberOrder.js - {"classMemberOrder":"private first"} format 1`] = `
====================================options=====================================
classMemberOrder: "private first"
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class TestClass {
    publicFieldFirst = 'Public field first'; // Public field
    #privateField = 'Private Field'; // Simulated private field using naming convention
    publicFieldLast = 'Public field last'; // Public field
    #privateFieldLast = 'Private Field Last'; // Simulated private field

    constructor() {
        this.publicFieldConstructor = 'Public field set in constructor';
        this._privateFieldConstructor = 'Private field set in constructor';
        console.log('Constructor');
    }
}
=====================================output=====================================
class TestClass {
  #privateField = "Private Field"; // Simulated private field using naming convention
  #privateFieldLast = "Private Field Last"; // Simulated private field

  publicFieldFirst = "Public field first"; // Public field
  publicFieldLast = "Public field last"; // Public field
  constructor() {
    this.publicFieldConstructor = "Public field set in constructor";
    this._privateFieldConstructor = "Private field set in constructor";
    console.log("Constructor");
  }
}

================================================================================
`;

exports[`classMemberOrder.js - {"classMemberOrder":"public first"} format 1`] = `
====================================options=====================================
classMemberOrder: "public first"
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class TestClass {
    publicFieldFirst = 'Public field first'; // Public field
    #privateField = 'Private Field'; // Simulated private field using naming convention
    publicFieldLast = 'Public field last'; // Public field
    #privateFieldLast = 'Private Field Last'; // Simulated private field

    constructor() {
        this.publicFieldConstructor = 'Public field set in constructor';
        this._privateFieldConstructor = 'Private field set in constructor';
        console.log('Constructor');
    }
}
=====================================output=====================================
class TestClass {
  publicFieldFirst = "Public field first"; // Public field
  publicFieldLast = "Public field last"; // Public field
  constructor() {
    this.publicFieldConstructor = "Public field set in constructor";
    this._privateFieldConstructor = "Private field set in constructor";
    console.log("Constructor");
  }
  #privateField = "Private Field"; // Simulated private field using naming convention
  #privateFieldLast = "Private Field Last"; // Simulated private field
}

================================================================================
`;
